From 3e31ab440d4494192ed01ff45f77d3de0fbe5bc2 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Wed, 2 Aug 2017 23:28:55 +0200
Subject: [PATCH 1195/1936] Make sure the bank view is refreshed when
 rescanning the banks. Also ensure the bankpos which now might be invalid is
 reset to zero.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Misc/MiddleWare.cpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 6a031457..5b5c1cd8 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -889,13 +889,23 @@ extern const rtosc::Ports bankPorts;
 const rtosc::Ports bankPorts = {
     {"rescan:", 0, 0,
         rBegin;
+        impl.bankpos = 0;
         impl.rescanforbanks();
         //Send updated banks
         int i = 0;
         for(auto &elm : impl.banks)
             d.reply("/bank/bank_select", "iss", i++, elm.name.c_str(), elm.dir.c_str());
         d.reply("/bank/bank_select", "i", impl.bankpos);
-
+        if (i > 0) {
+            impl.loadbank(impl.banks[0].dir);
+
+            //Reload bank slots
+            for(int i=0; i<BANK_SIZE; ++i) {
+                d.reply("/bankview", "iss",
+                    i, impl.ins[i].name.c_str(),
+                    impl.ins[i].filename.c_str());
+            }
+        }
         rEnd},
     {"bank_list:", 0, 0,
         rBegin;
-- 
2.47.0

